cmd/notation/plugin/install.go

```diff
@@ -0,0 +1,329 @@
+	}
+	// checksum check
+	if inputCheckSum != "" {
+		if err := notationplugin.ValidateCheckSum(inputPath, inputCheckSum); err != nil {
+			return fmt.Errorf("failed to install the plugin: %w", err)
+		}
+	}
+	// install the plugin based on file type
+	fileType, err := osutil.DetectFileType(inputPath)
+	if err != nil {
+		return fmt.Errorf("failed to install the plugin: %w", err)
+	}
+	switch fileType {
+	case notationplugin.MediaTypeZip:
+		if err := installPluginFromZip(ctx, inputPath, force); err != nil {
+			return fmt.Errorf("failed to install the plugin: %w", err)
+		}
+		return nil
+	case notationplugin.MediaTypeGzip:
+		// when file is gzip, require to be tar
+		if err := installPluginFromTarGz(ctx, inputPath, force); err != nil {
+			return fmt.Errorf("failed to install the plugin: %w", err)
+		}
+		return nil
+	default:
+		return errors.New("failed to install the plugin: invalid file format. Only support .tar.gz and .zip")
+	}
+}
+
+// installPluginFromZip extracts a plugin zip file, validates and
+// installs the plugin
+func installPluginFromZip(ctx context.Context, zipPath string, force bool) error {
+	logger := log.GetLogger(ctx)
+	archive, err := zip.OpenReader(zipPath)
+	if err != nil {
+		return err
+	}
+	defer archive.Close()
+	// require one and only one file with name in the format
+	// `notation-{plugin-name}`
+	for _, f := range archive.File {
+		if !f.Mode().IsRegular() || strings.Contains(f.Name, "..") {
+			continue
+		}
+		// validate and get plugin name from file name
+		pluginName, err := notationplugin.ExtractPluginNameFromFileName(f.Name)
+		if err != nil {
+			logger.Infoln(err)
+			continue
+		}
+		fileInArchive, err := f.Open()
+		if err != nil {
+			return err
+		}
+		defer fileInArchive.Close()
+		return installPluginExecutable(ctx, f.Name, pluginName, fileInArchive, force)
+	}
+	return fmt.Errorf("no valid plugin file was found in %s. Plugin file name must in format notation-{plugin-name}", zipPath)
+}
+
+// installPluginFromTarGz extracts and untar a plugin tar.gz file, validates and
+// installs the plugin
+func installPluginFromTarGz(ctx context.Context, tarGzPath string, force bool) error {
+	logger := log.GetLogger(ctx)
+	rc, err := os.Open(tarGzPath)
+	if err != nil {
+		return err
+	}
+	defer rc.Close()
+	decompressedStream, err := gzip.NewReader(rc)
+	if err != nil {
+		return err
+	}
+	defer decompressedStream.Close()
+	tarReader := tar.NewReader(decompressedStream)
+	// require one and only one file with name in the format
+	// `notation-{plugin-name}`
+	for {
+		header, err := tarReader.Next()
+		if err != nil {
+			if err == io.EOF {
+				break
+			}
+			return err
+		}
+		if !header.FileInfo().Mode().IsRegular() || strings.Contains(header.Name, "..") {
+			continue
+		}
+		// validate and get plugin name from file name
+		pluginName, err := notationplugin.ExtractPluginNameFromFileName(header.Name)
+		if err != nil {
+			logger.Infoln(err)
```

Comment:
It is possible that the tar contains a LICENSE file. In that case, the log will show
> Info: invalid plugin file name. Plugin file name requires format notation-{plugin-name}, but got LICENSE 

It is an expected file so can we update the log to make it clear.

Reply:
There could be various expected files in a tar, for example, `root.cert`, `LICENSE`, and other user customized files. So only checking the `LICENSE` file doesn't seem to be very helpful? That's why I used a `logger.Infoln` to log the error instead of treating it as a real error.

Reply:
Updated the logging message content to be clearer.